scipy/stats/tests/test_stats.py

```diff
@@ -4480,6 +4494,67 @@ def test_constant_input(self, a, b, expected):
             f, p = stats.f_oneway(a, b)
             assert f, p == expected
 
+    def test_2d_inputs(self):
+        a = np.array([[1, 4, 3, 3],
+                      [2, 5, 3, 3],
+                      [3, 6, 3, 3],
+                      [2, 3, 3, 3],
+                      [1, 4, 3, 3]])
+        b = np.array([[3, 1, 5, 3],
+                      [4, 6, 5, 3],
+                      [4, 3, 5, 3],
+                      [1, 5, 5, 3],
+                      [5, 5, 5, 3],
+                      [2, 3, 5, 3],
+                      [8, 2, 5, 3],
+                      [2, 2, 5, 3]])
+        c = np.array([[4, 3, 4, 3],
+                      [4, 2, 4, 3],
+                      [5, 4, 4, 3],
+                      [5, 4, 4, 3]])
+
+        with assert_warns(stats.F_onewayConstantInputWarning):
+            f, p = stats.f_oneway(a, b, c)
+
+        # Verify the result computed with the 2d arrays matches the result
+        # from calling f_oneway individually on each slice.
+        for j in [0, 1]:
+            fj, pj = stats.f_oneway(a[:, j], b[:, j], c[:, j])
+            assert_allclose(f[j], fj, rtol=1e-14)
+            assert_allclose(p[j], pj, rtol=1e-14)
+        for j in [2, 3]:
+            with assert_warns(stats.F_onewayConstantInputWarning):
+                fj, pj = stats.f_oneway(a[:, j], b[:, j], c[:, j])
+                assert_equal(f[j], fj)
+                assert_equal(p[j], pj)
+
+    def test_length0_error(self):
+        # Require at least one value in each group.
+
+        with assert_warns(stats.F_onewayBadInputSizesWarning):
```

Comment:
Seems like both the tests inside the separate context managers could be factored out into `@pytest.mark.parametrize` tuple inputs.

Reply:
I'll just split this test into two tests.

Reply:
that works too